lib/plugins/aws/package/compile/events/kafka.js

```diff
@@ -132,6 +124,37 @@ class AwsCompileKafkaEvents {
       functionObj.events.forEach((event) => {
         if (!event.kafka) return;
 
+        if (!event.kafka.accessConfigurations || event.kafka.accessConfigurations.length === 0) {
+          throw new ServerlessError(
+            `You must provide at least one accessConfiguration for function: ${functionName}`,
+            'FUNCTION_KAFKA_ACCESS_CONFIGURATION_EMPTY'
+          );
+        }
+
+        const {
+          accessConfigurations: {
+            vpcSecurityGroup,
+            vpcSubnet,
+            clientCertificateTlsAuth,
+            serverRootCaCertificate,
+          },
+        } = event.kafka;
+
+        if ((vpcSecurityGroup && !vpcSubnet) || (vpcSubnet && !vpcSecurityGroup)) {
```

Comment:
@medikoo @pgrzesik I've shifted the validation logic from the schema to here. Hope this is ok!

Reply:
It looks great here 👍